tests/Unit/FeedXMLGenerationTest.php

```diff
@@ -389,6 +399,55 @@ public function testPropertyAdditionalImageLinkXML() {
 		$this->assertEquals( '<g:additional_image_link><![CDATA[http://example.org/wp-content/uploads/product_image_1.png,http://example.org/wp-content/uploads/product_image_2.png]]></g:additional_image_link>', $xml );
 	}
 
+	/**
+	 * @group feed
+	 */
+	public function testAttributesConditionXML() {
+		$method = ( new ReflectionClass( ProductsXmlFeed::class ) )->getMethod( 'get_attributes_xml' );
+		$method->setAccessible( true );
+
+		$product = WC_Helper_Product::create_simple_product();
+		$xml = $method->invoke( null, $product, '\t' );
+		// No attributes set, output should be empty.
+		$this->assertEquals( '', $xml );
+
+		$condition         = new Condition( 'new' );
+		$attribute_manager = AttributeManager::instance();
+		$attribute_manager->update( $product, $condition );
+
+		$xml = $method->invoke( null, $product, '' );
+		// Condition attribute was set, we should see it in the output.
+		$this->assertEquals( '<g:condition>new</g:condition>' . PHP_EOL, $xml );
+	}
+
+	/**
+	 * @group feed
+	 *
+	 * Since GoogleCategory is not validated we would end up with yet another test like testAttributesConditionXML.
+	 * To make this a bit more useful this will be more of an integration test.
+	 */
+	public function testAttributesGoogleCategoryXML() {
+		$method = ( new ReflectionClass( ProductsXmlFeed::class ) )->getMethod( 'get_attributes_xml' );
+		$method->setAccessible( true );
+
+		$product = WC_Helper_Product::create_simple_product();
+		$xml = $method->invoke( null, $product, '\t' );
+		// No attributes set, output should be empty.
+		$this->assertEquals( '', $xml );
+
+		$full_category_name_method = new ReflectionMethod( GoogleCategorySearch::class, 'full_category_name' );
```

Comment:
Bit of a mixed feeling on this. I think the main purpose of this unit test is to test the generation of the product feed. Converting the category name and confirming the search functionality works should have a unit test of it's own, maybe another PR. To generate the feed we just get the full category name which is saved in the product meta, so we can just simply set the product meta when creating the product and confirm it's output in the feed.

Reply:
I also thought about this - hence the comment. Having separate tests for this makes 100% sense, but I thought that having a bit more coverage, even a bit sideways is still good.